src/openloco/window.cpp

```diff
@@ -387,16 +387,19 @@ namespace openloco::ui
 
     // 0x00459E54
     // TODO: needs expansion in terms of (output) parameters.
-    static void get_map_coordinates_from_pos(int32_t screenX, int32_t screenY, int32_t flags, int16_t* x, int16_t* y)
+    viewport_interaction::InteractionItem get_map_coordinates_from_pos(int32_t screenX, int32_t screenY, int32_t flags, int16_t* x, int16_t* y, void** dest)
     {
         registers regs;
         regs.ax = screenX;
-        regs.cx = screenY;
+        regs.bx = screenY;
```

Comment:
@AaronVanGeffen Does this change fix your 'zoom to cursor' woes?

Reply:
Yes!! Just tested this and I'm baffled — this is great news! Thank you so much.

Shall we tackle it in a different PR? We could enable it by default and all that, too.

Reply:
Yeah, this PR contains lots of stuf that would probably be better extracted